plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergAbfsConnectorSmokeTest.java

```diff
@@ -0,0 +1,154 @@
+
+        this.hiveMinioDataLake = closeAfterClass(new HiveMinioDataLake(
+                "test-bucket",
+                ImmutableMap.of("/etc/hadoop/conf/core-site.xml", hadoopCoreSiteXmlTempFile.normalize().toAbsolutePath().toString()),
+                HiveHadoop.HIVE3_IMAGE));
+        this.hiveMinioDataLake.start();
+
+        return IcebergQueryRunner.builder()
+                .setIcebergProperties(
+                        ImmutableMap.<String, String>builder()
+                                .put("iceberg.file-format", format.name())
+                                .put("iceberg.catalog.type", "HIVE_METASTORE")
+                                .put("hive.metastore.uri", "thrift://" + hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint())
+                                .put("hive.metastore-timeout", "1m") // read timed out sometimes happens with the default timeout
+                                .put("hive.azure.abfs-storage-account", account)
+                                .put("hive.azure.abfs-access-key", accessKey)
+                                .put("iceberg.register-table-procedure.enabled", "true")
+                                .buildOrThrow())
+                .setSchemaInitializer(
+                        SchemaInitializer.builder()
+                                .withSchemaName(schemaName)
+                                .withClonedTpchTables(REQUIRED_TPCH_TABLES)
+                                .withSchemaProperties(Map.of("location", "'" + formatAbfsUrl(container, account, bucketName) + schemaName + "'"))
+                                .build())
+                .build();
+    }
+
+    @Override
+    protected String createSchemaSql(String schemaName)
+    {
+        return "CREATE SCHEMA IF NOT EXISTS " + schemaName + " WITH (location = '" + formatAbfsUrl(container, account, bucketName) + schemaName + "')";
+    }
+
+    @Test
+    @Override
+    public void testRenameSchema()
+    {
+        assertQueryFails(
+                format("ALTER SCHEMA %s RENAME TO %s", schemaName, schemaName + randomNameSuffix()),
+                "Hive metastore does not support renaming schemas");
+    }
+
+    @Override
+    protected void dropTableFromMetastore(String tableName)
+    {
+        HiveMetastore metastore = new BridgingHiveMetastore(
+                testingThriftHiveMetastoreBuilder()
+                        .metastoreClient(hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint())
+                        .build());
+        metastore.dropTable(schemaName, tableName, false);
+        assertThat(metastore.getTable(schemaName, tableName)).isEmpty();
+    }
+
+    @Override
+    protected String getMetadataLocation(String tableName)
+    {
+        HiveMetastore metastore = new BridgingHiveMetastore(
+                testingThriftHiveMetastoreBuilder()
+                        .metastoreClient(hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint())
+                        .build());
```

Comment:
How about moving `HiveMetastore metastore` to the field?

Reply:
These two methods of copied into a couple different classes, maybe we clean up separately?